crates/cansat-lora/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+#![no_std]
+
+use embedded_hal::{self, delay, nb, serial};
+
+pub enum Error<SerialError>
+where
+    SerialError: serial::Error,
+{
+    Delay,
+    ResponseError,
+    Serial(SerialError),
+}
+
+pub struct Lora<Serial> {
+    serial: Serial,
+}
+
+impl<Serial> Lora<Serial>
+where
+    Serial: serial::nb::Write<u8> + serial::nb::Read<u8>,
+{
+    pub fn new(serial: Serial) -> Self {
+        Self { serial }
+    }
+
+    fn write_all(&mut self, cmd: &[u8]) -> Result<(), Error<Serial::Error>> {
+        for &b in cmd {
+            nb::block!(self.serial.write(b)).map_err(Error::Serial)?;
+        }
+
+        Ok(())
+    }
+
+    fn read_all(&mut self, buffer: &mut [u8]) -> Result<usize, Error<Serial::Error>> {
+        let mut ptr = 0;
+
+        loop {
+            let b = nb::block!(self.serial.read()).map_err(Error::Serial)?;
+
+            buffer[ptr] = b;
+            ptr = (ptr + 1) % buffer.len();
```

Comment:
I'm not sure, but this can lead to reading invalid future responses. Each response ends with `\r\n`, so for example, we have `1234\r\n` response and buffer of size `3`. We read `123` then return error. If now we call `read_all` again, then it is possible that we will read `4\r\n` and stop because of `\r\n`. 

I think there should be some loop to read to the end of message. Idk how this protocol works, maybe just delay caused by sending message is enough to skip unread data.

Reply:
@PatrykPaluch Great point! I'm pretty sure serial is buffered, so I think you are correct. This is somethng we will have to test with hardware.

Reply:
Just tested it and it in fact it's buffered. It gets automatically cleared after ~10ms though, so it probably wouldn't be a problem in practice, but let's just drain the buffer just in case.
```rust
fn drain(&mut self) -> Result<(), Error> {
    loop {
        match self.serial.read() {
            Ok(_) => (),
            Err(nb::Error::WouldBlock) => break,
            Err(nb::Error::Other(E)) => return Err(Error::Serial(E)),
        }
    }
    Ok(())
}
```